sos/report/plugins/bcache.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2021, Canonical ltd
+# Ponnuvel Palaniyappan <ponnuvel.palaniyappan@canonical.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin, SoSPredicate
+
+
+class Bcache(Plugin, IndependentPlugin):
+
+    short_desc = 'Bcache statistics'
+
+    plugin_name = 'bcache'
+    profiles = ('storage', 'hardware')
+    files = ('/sys/fs/bcache',)
+
+    def setup(self):
+
+        # Caution: reading /sys/fs/bcache/*/cache0/priority_stats is known
+        # to performance degradation on some kernels. Needs care if that's
```

Comment:
A small nit, recommend changing to:
`degrade performance on older kernels.`

I'm suggesting this change just to be clear that the issue has been fixed in recent kernels.

Reply:
Hi Dan,
My understanding is that performance is much improved with that fix, but it still has some impact. I am referring this message in that thread: https://lkml.org/lkml/2019/8/14/636

Reply:
Ah, you are right. There is still a minor performance hit due to locking.